rust/agents/relayer/src/relayer.rs

```diff
@@ -403,6 +424,7 @@ impl Relayer {
     async fn run_interchain_gas_payment_sync(
         &self,
         origin: &HyperlaneDomain,
+        mut rxs: HashMap<HyperlaneDomain, BroadcastReceiver<H512>>,
```

Comment:
thinking about it more, because we clone the entire hashmap for each origin chain, iiuc we end up creating n^2 broadcast receivers (instead of `n`, where n = the number of origin chains) when running all the IGP syncs. Because they don't get cleaned up (`rxs` is moved into the task), the receivers for all the other origins won't ever read from the channel, so the data will stick around in memory https://docs.rs/tokio/latest/tokio/sync/broadcast/index.html#lagging. So we should def change this to just the broadcast receiver

We could also maybe just make it a non-optional receiver too if we really wanna make sure that we always have a receiver here?

Reply:
Great point, changed to passing a single receiver. The receiver is intentionally non-optional, since e.g. the message sync won't be receving anything, but just broadcasting